### ISSUE TITLE
Workspace symbols not behaving as expected

### ISSUE THREAD

**Issue by nomasprime** [castwide/solargraph#328]
Using NeoVim's built-in LSP client with the following map:

```
nnoremap <silent> gW    <cmd>lua vim.lsp.buf.workspace_symbol()<CR>
```

I get prompted for a query, hit enter to get all results but then nothing happens.

When I enter a query, eg, "scan", the first result is a method `scan` the rest seem unrelated, eg, a property `count`.

**Comment by nomasprime** [comment]
Might be that the [docs](https://neovim.io/doc/user/lsp.html) aren't explicit enough. Could be interpreted that when no query is sent all symbols are returned but this could create a lot of overhead. Makes sense that results would only be returned if a non-empty query was sent.

I also discovered queries are fuzzy matched so that probably explains why they seem unrelated at first glance.

If results are only supposed to be returned when a non-empty query is sent, this issue can be closed.

**Comment by castwide** [comment]
It looks like the [specification](https://microsoft.github.io/language-server-protocol/specification#workspace_symbol) also suggests returning all symbols for an empty query. Seems reasonable to me. I'll add it to the next patch release. You can keep this issue open if you want a notification when it's done.

**Comment by nomasprime** [comment]
Cool. Thanks @castwide.

**Comment by castwide** [comment]
Released in v0.39.9.
